### DISCUSSION TITLE
Sub-Package config support in monorepo

### DISCUSSION THREAD

**Discussion by samjcombs** [renovatebot/renovate#20133] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.105.6

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.5.2

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have a monorepo with a top level renovate config that pulls in a shared config thats used in many repos.  

In this monorepo, I'd like to be able to put a package level renovate config in a certain group of packages and have them inherit a config file thats used in another set of packages. 

So kindof like this: 
```
monorepo
|
+-- renovate.json5 (top-level config)
|
+-- package-group-1
|   |
|   +-- package-1 (inherits config from top-level config)
|   |
|   +-- package-2 (inherits config from top-level config)
|
+-- package-group-2
    |
    +-- package-3
    |   |
    |   +-- renovate.json5 (inherits specific config from elsewhere) <- IS THIS POSSIBLE?
    |  
    +-- package-4
        |
        +-- renovate.json5 (inherits specific config from elsewhere) <- IS THIS POSSIBLE?
```

### What you tried so far.

I tried putting package level configs in but it doesnt recognize them

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by secustor** [comment, marked as answer]
This is currently not supported.    
Renovate will only detected configuration in specific locations. https://docs.renovatebot.com/getting-started/installing-onboarding/#configuration-location

**Reply by samjcombs** [reply]
Thanks, is it possible to extend from one preset for certain directories?  like specify a few directories in which to extend from a different preset than the default?

**Reply by secustor** [reply]
You could use a [`matchPaths`](https://docs.renovatebot.com/configuration-options/#matchpaths) in combination with `extends` 

```json
{
  "packageRules": [
    {
       "matchPaths": ["myfolder/**"],
       "extends": [
          "group:all"
       ]
    }
  ]
}
```

But to be honest, I have not used extends this way. So your mileage may vary.

**Comment by viceice** [comment]
you can create package rules with match paths or files to apply to specific folders .
but it still needs to be one config file per repo

**Reply by samjcombs** [reply]
cool, i wonder if i could inherit settings in a packageRule from another config?   likely not... this though is a great solution thanks
